### ISSUE TITLE
avm1: Some properties seem to only need a setter

### ISSUE THREAD

**Issue by Toad06** [ruffle-rs/ruffle#10573]
This affects at least the `MovieClip.focusEnabled` property. Consider the following sample [(SWF available here)](https://github.com/ruffle-rs/ruffle/files/11163341/focusEnabled.zip):

```
_root.createEmptyMovieClip("mc", _root.getNextHighestDepth());

trace("mc.focusEnabled: " + mc.focusEnabled);
trace("Selection.setFocus(mc): " + Selection.setFocus(mc));
trace("Selection.getFocus(): " + Selection.getFocus());

mc.focusEnabled = "whatever";
trace("mc.focusEnabled = 'whatever'");

trace("mc.focusEnabled: " + mc.focusEnabled);
trace("Selection.setFocus(mc): " + Selection.setFocus(mc));
trace("Selection.getFocus(): " + Selection.getFocus());

delete mc.focusEnabled;
trace("delete mc.focusEnabled");

trace("mc.focusEnabled: " + mc.focusEnabled);
trace("Selection.getFocus(): " + Selection.getFocus());
```

In Flash Player, this traces the following data:

```
mc.focusEnabled: undefined
Selection.setFocus(mc): false
Selection.getFocus(): null

mc.focusEnabled = 'whatever'

mc.focusEnabled: whatever
Selection.setFocus(mc): true
Selection.getFocus(): _level0.mc

delete mc.focusEnabled

mc.focusEnabled: undefined
Selection.getFocus(): _level0.mc
```

---

So it appears that the `focusEnabled` getter always returns the value set by the script (deleting the property also works), like any classic properties (using `delete` also works). In Ruffle, we assume this value to always be a boolean:
https://github.com/ruffle-rs/ruffle/blob/7a6e6cf21475225dc6066381eaf1404d43d4d726/core/src/avm1/globals/movie_clip.rs#L1429-L1434

The `focusEnabled` setter has an impact on `Selection.setFocus`, as Ruffle already implements:
https://github.com/ruffle-rs/ruffle/blob/7a6e6cf21475225dc6066381eaf1404d43d4d726/core/src/avm1/globals/movie_clip.rs#L1436-L1446
https://github.com/ruffle-rs/ruffle/blob/7a6e6cf21475225dc6066381eaf1404d43d4d726/core/src/avm1/globals/selection.rs#L143-L146

---

However, only mentioning a setter in `property()` doesn't seem allowed:
https://github.com/ruffle-rs/ruffle/blob/7a6e6cf21475225dc6066381eaf1404d43d4d726/core/src/avm1/globals/movie_clip.rs#L110

**Comment by Herschel** [comment]
The aren't actually built-in properties of `MovieClip`, despite being listed as such in the AS documentation:
```
trace(MovieClip.prototype.hasOwnProperty("focusEnabled")); // false
trace(someClip.hasOwnProperty("focusEnabled")); // false, you had set it
```

So to be accurate, they shouldn't be listed in the `PROTO_DECLS` list at all. Instead, methods such as `is_focusable` should check `get_property("focusEnabled")` on demand, which would be undefined -> false in the normal case where it hasn't been set.

**Comment by relrelb** [comment]
@Toad06 Can this issue be closed now, after #10597, #10865, #10971 got merged? Or there are still some remaining properties?

**Comment by Toad06** [comment]
There are remaining properties that show the same behavior (like `tabEnabled` or `tabChildren`)... but we don't implement them yet. :) So I think it should be fine to close this issue.
